src/piplint/__init__.py

```diff
@@ -187,18 +188,20 @@ def valid_version(version, compare, r_version):
         print("%s%s%s\n" % (colour.WARNING,
                             "\n".join(sorted(unknown_reqs_with_versions)),
                             colour.ENDC))
+        if error_on_extras:
+            code = 2
```

Comment:
My first inclination was to keep it as a separate path. I don't have a strong justification for this, only that one might want to handle the two cases differently.